keras/layers/convolutional_recurrent.py

```diff
@@ -1,172 +1,585 @@
+            - specify `shuffle=False` when calling fit().
 
         To reset the states of your model, call `.reset_states()` on either
         a specific layer, or on your entire model.
+
+    # Note on specifying the initial state of RNNs
+        You can specify the initial state of RNN layers symbolically by
+        calling them with the keyword argument `initial_state`. The value of
+        `initial_state` should be a tensor or list of tensors representing
+        the initial state of the RNN layer.
+
+        You can specify the initial state of RNN layers numerically by
+        calling `reset_states` with the keyword argument `states`. The value of
+        `states` should be a numpy array or list of numpy arrays representing
+        the initial state of the RNN layer.
+
+    # Note on passing external constants to RNNs
+        You can pass "external" constants to the cell using the `constants`
+        keyword argument of `RNN.__call__` (as well as `RNN.call`) method. This
+        requires that the `cell.call` method accepts the same keyword argument
+        `constants`. Such constants can be used to condition the cell
+        transformation on additional static inputs (not changing over time),
+        a.k.a. an attention mechanism.
     """
 
-    def __init__(self, filters,
-                 kernel_size,
-                 strides=(1, 1),
-                 padding='valid',
-                 data_format=None,
-                 dilation_rate=(1, 1),
+    def __init__(self, cell,
                  return_sequences=False,
+                 return_state=False,
                  go_backwards=False,
                  stateful=False,
                  **kwargs):
-        super(ConvRecurrent2D, self).__init__(**kwargs)
-        self.filters = filters
-        self.kernel_size = conv_utils.normalize_tuple(kernel_size, 2, 'kernel_size')
-        self.strides = conv_utils.normalize_tuple(strides, 2, 'strides')
-        self.padding = conv_utils.normalize_padding(padding)
-        self.data_format = conv_utils.normalize_data_format(data_format)
-        self.dilation_rate = conv_utils.normalize_tuple(dilation_rate, 2, 'dilation_rate')
+        if not hasattr(cell, 'call'):
+            raise ValueError('`cell` should have a `call` method. '
+                             'The RNN was passed:', cell)
+        if not hasattr(cell, 'state_size'):
+            raise ValueError('The RNN cell should have '
+                             'an attribute `state_size` '
+                             '(tuple of integers, '
+                             'one integer per RNN state).')
+        super(ConvRNN2D, self).__init__(**kwargs)
+        self.cell = cell
         self.return_sequences = return_sequences
+        self.return_state = return_state
         self.go_backwards = go_backwards
         self.stateful = stateful
+
+        # Masking is only supported in Theano
```

Comment:
Is this still true? What's the underlying reason?

Reply:
My bad, it was in the docstring of `ConvRecurrent2D`, I didn't check if it was still true or not.